tests/framework/extensions/clusters/clusters.go

```diff
@@ -201,6 +206,47 @@ func NewK3SRKE2ClusterConfig(clusterName, namespace, cni, cloudCredentialSecretN
 		WorkerConcurrency:        "10%",
 		WorkerDrainOptions:       rkev1.DrainOptions{},
 	}
+	clusterAgentDeploymentCustomization := &apisV1.AgentDeploymentCustomization{}
+	if advancedOptions.ClusterAgentCustomization.OverrideResourceRequirements != nil {
+		clusterAgentOverrides := ResourceConfigHelper(advancedOptions.ClusterAgentCustomization.OverrideResourceRequirements)
+		clusterAgentDeploymentCustomization.OverrideResourceRequirements = &clusterAgentOverrides
+	}
+	if advancedOptions.ClusterAgentCustomization.AppendTolerations != nil {
+		v1ClusterTolerations := []corev1.Toleration{}
+		for _, t := range advancedOptions.ClusterAgentCustomization.AppendTolerations {
+			v1ClusterTolerations = append(v1ClusterTolerations, corev1.Toleration{
+				Key:      t.Key,
+				Operator: corev1.TolerationOperator(t.Operator),
+				Value:    t.Value,
+				Effect:   corev1.TaintEffect(t.Effect),
+			})
+		}
+		clusterAgentDeploymentCustomization.AppendTolerations = v1ClusterTolerations
+	}
+	if advancedOptions.ClusterAgentCustomization.OverrideAffinity != nil {
+		clusterAgentDeploymentCustomization.OverrideAffinity = AgentAffinityConfigHelper(advancedOptions.ClusterAgentCustomization.OverrideAffinity)
+	}
```

Comment:
agreed will make it look cleaner and easier to follow

Reply:
so, since these fields override, adding them if they are empty will cause problems (especially for the affinity field). I could add something like 
```
	if (&corev1.ResourceRequirements{}) != &agentOverrides {
		//set the values here
	}
```
but it would still require some conditionals.